plasmapy/utils/decorators/deprecation.py

```diff
@@ -0,0 +1,16 @@
+__all__ = ["deprecated"]
+
+import astropy
+
+from functools import partial
+
+from plasmapy.utils.exceptions import PlasmaPyDeprecationWarning
+
+deprecated = partial(
+    astropy.utils.decorators.deprecated,
+    warning_type=PlasmaPyDeprecationWarning,
+)
+"""
+A wrapper of `astropy.utils.decorators.deprecated` that by default assumes
+a warning type of `~plasmapy.utils.exceptions.PlasmaPyDeprecationWarning`.
+"""
```

Comment:
So...While I like `functools.partial`, it does cause some documentation issues here.  If you go look at the RTD builds, `deprecated` is showing up under "Variables & Attributes."  This is because `deprecated` is missing [all the attributes that would make it look like a function](https://github.com/python/cpython/blob/eaede0ded72e67cee4a91c086847d54cb64ca74c/Lib/inspect.py#L277-L288).  This makes me side on that `functools.partial` should be restricted to being used in code and not for outward functionality/API.

AS a work around, here's my suggested replacement.  It's not as elegant, but it does what it should.

```suggestion
__all__ = ["deprecated"]

import inspect

from astropy.utils.decorators import deprecated as astropy_deprecated

from plasmapy.utils.exceptions import PlasmaPyDeprecationWarning


def deprecated(*args, warning_type=PlasmaPyDeprecationWarning, **kwargs):
    """
    A wrapper of `astropy.utils.decorators.deprecated` that by default assumes
    a warning type of `~plasmapy.utils.exceptions.PlasmaPyDeprecationWarning`.
    """
    return astropy_deprecated(*args, warning_type=warning_type, **kwargs)


# override deprecated's signature
asig = inspect.signature(astropy_deprecated)
new_default = asig.parameters["warning_type"].replace(
    default=PlasmaPyDeprecationWarning
)
new_sig = dict(asig.parameters)
new_sig["warning_type"] = new_default
new_sig = inspect.Signature(
    parameters=tuple(new_sig.values()), return_annotation=asig.return_annotation,
)
deprecated.__signature__ = new_sig
"""
```

Reply:
Thanks — I think we can do that.  

> This makes me side on that `functools.partial` should be restricted to being used in code and not for outward functionality/API.

This seems quite sensible to me.  If we end up using `functools.partial` a lot then we could put the signature override into a function that implements your suggestion, but that should wait until we have a demonstrated need for it.